bot/resources/tags/listcomps.md

```diff
@@ -1,14 +1,19 @@
-Do you ever find yourself writing something like:
+Do you ever find yourself writing something like this?
 ```py
-even_numbers = []
-for n in range(20):
-    if n % 2 == 0:
-        even_numbers.append(n)
+>>> squares = []
+>>> for n in range(5):
+...    squares.append(n ** 2)
+[0, 1, 4, 9, 16]
 ```
-Using list comprehensions can simplify this significantly, and greatly improve code readability. If we rewrite the example above to use list comprehensions, it would look like this:
+Using list comprehensions can make this both shorter and more readable. As a list comprehension, the same code would look like this:
 ```py
-even_numbers = [n for n in range(20) if n % 2 == 0]
+>>> [n ** 2 for n in range(5)]
+[0, 1, 4, 9, 16]
+```
+List comprehensions also get an `if` statement:
+```python
+>>> [n ** 2 for n in range(5) if n % 2 == 0]
+[0, 4, 16]
 ```
-This also works for generators, dicts and sets by using `()` or `{}` instead of `[]`.
```

Comment:
Is there a reason to remove this piece of information? We may want to say that it works in a similar way, instead of that it work out of the box, for the other types. From using the tag in the community, I've found this to be a nice addition to the tag, by showing them that there's more to the comps than just list comps. It's a pretty nice fact.

Reply:
Other types of comprehensions have differences that people need to know about in order to use them and given that it would take too many words to describe those differences, I think we shouldn't refer to them at all. I think this is especially true for generator expressions, as I think most people learn list and dict comprehensions (and maybe set comprehensions as well) before they learn about lazy iteration.

`!dictcomp` is another tag.

Reply:
I think you're right that they have quite a few differences that you need to know. However, this is something that I actually found very useful when reading the tag a while back. List comps are pretty common, so mentioning that there are similar things and underlining the fact that there are differences and that they'll need to do some further resource is crucial. At the same time, I see this as a great opportunity to sneak in some "you didn't know you wanted to know this, but you really want to know this" kind of content.